lib/theme-api-client.js

```diff
@@ -0,0 +1,174 @@
+        url: `${options.apiHost}/stores/${options.storeHash}/v3/themes/jobs/${options.jobId}`,
+    };
+
+    request(opts, (err, res, payload) => {
+        let error;
+
+        if (err) {
+            err.name = "JobCompletionStatusCheckError"
+            return callback(err);
+        }
+
+        if (res.statusCode !== 200 || payload.data.status === 'FAILED') {
+            error = new Error('Job Failed');
+            error.name = "JobCompletionStatusCheckError";
+            return callback(error);
+        }
+
+        if (payload.data.status !== 'COMPLETED') {
+            error = new Error(payload.data.percent_complete);
+            error.name = "JobCompletionStatusCheckPendingError";
+            return callback(error);
+        }
+
+        callback(null, Object.assign({}, options, {
+            themeId: payload.data.result.theme_id,
+        }));
+    });
+}
+
+function getThemes(options, callback) {
+    Wreck.get(`${options.apiHost}/stores/${options.storeHash}/v3/themes`, {
+        headers: {
+            'x-auth-client': options.clientId,
+            'x-auth-token': options.accessToken,
+        },
+        json: true,
+    }, (err, res, payload) => {
+        if (err) {
+            err.name = 'ThemesRetrievalError';
+            return callback(err);
+        }
+
+        callback(null, Object.assign({}, options, {
+            themes: payload.data,
+        }));
+    });
+}
+
+function getVariationsByThemeId(options, callback) {
+    Wreck.get(`${options.apiHost}/stores/${options.storeHash}/v3/themes/${options.themeId}`, {
+        json: true,
+        headers: {
+            'x-auth-client': options.clientId,
+            'x-auth-token': options.accessToken,
+        },
+    }, (err, res, payload) => {
+        if (err) {
+            err.name = "VariationsRetrievalError";
+            return callback(err);
+        }
+
+        callback(null, Object.assign({}, options, {
+            variations: payload.data.variations,
+        }));
+    });
+}
+
+function postTheme(options, callback) {
+    const opts = {
+        formData: {
+            file: {
+                options: {
+                    contentType: null,
+                    filename: path.basename(options.bundleZipPath),
+                },
+                value: fs.createReadStream(options.bundleZipPath),
+            },
+        },
+        headers: {
+            'cache-control': 'no-cache',
+            'content-type': 'multipart/form-data',
+            'x-auth-token': options.accessToken,
+            'x-auth-client': options.clientId,
+        },
+        json: true,
+        method: 'POST',
+        url: `${options.apiHost}/stores/${options.storeHash}/v3/themes`,
+    };
+
+    request(opts, (error, response, payload) => {
+        if (error) {
+            return callback(error);
+        }
+
+        if (response.statusCode !== 201) {
+            if (response.statusCode === 502 && payload && /upload limit/.test(payload.title)) {
```

Comment:
@junedkazi - can we change the statusCode to something more relevant or include a code within the body?  i'm worried that the `title` field's value may change in the future.  502 doesn't work here because it's used for other scenarios.